### PR TITLE
Fix/208 correct discount voucher handling with partial payment transactions

### PR DESCRIPTION
**Pull request by pSpeck**
| Q             | A
| ------------- | ---
| Branch        | 6.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no 
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#208
| License       | MIT

### CODE REVIEW

File: src/ShopPaymentTransactionBundle/objects/TPkgShopPaymentTransactionManagerEndPoint.class.php [hunk 1]
```diff
@@ -223,6 +225,75 @@ protected function updateOrderPaidStatusBasedOnCurrentBalance()
...
+                return;
+            }
+            $this->changeVoucherUsedUp($voucher);
+            if (0 === $voucherUsedDifference) {
+                break;
+            }
+        }
+    }
+
+    private function changeVoucherUseValue(TAdbShopVoucherUse $voucherUse, float $newVoucherUsedValue): void
```
[the remark this answers is not in the document]
Comment (pSpeck) [reply src/ShopPaymentTransactionBundle/objects/TPkgShopPaymentTransactionManagerEndPoint.class.php]: changed

File: src/ShopPaymentTransactionBundle/objects/TPkgShopPaymentTransactionManagerEndPoint.class.php [hunk 2]
```diff
@@ -669,4 +726,35 @@ private function getOrderTotalOtherValue()
...
         return $dOrderTotal;
     }
+
+    /**
+     * @param $sTransactionType
+     * @param $dOrderTotal
+     *
+     * @return float|int
+     */
+    private function getTransactionOrderDataVoucherValue($sTransactionType, $dOrderTotal)
```
[the remark this answers is not in the document]
Comment (pSpeck) [reply src/ShopPaymentTransactionBundle/objects/TPkgShopPaymentTransactionManagerEndPoint.class.php]: changed

File: src/ShopPaymentTransactionBundle/objects/TPkgShopPaymentTransactionManagerEndPoint.class.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (umulmrum) [reply src/ShopPaymentTransactionBundle/objects/TPkgShopPaymentTransactionManagerEndPoint.class.php]: Argh I'm sorry, I misread the line - $voucherUse is of course correct, not $voucher.

**Comment by umulmrum** [comment]
Thanks @pSpeck

**Pull request merged**